src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecLoaderExtensions.cs

```diff
@@ -16,39 +16,84 @@
 
 using System;
 using System.IO;
+using System.Linq;
+using System.Reflection;
 using System.Text;
+using Nethermind.Config;
 using Nethermind.Logging;
 
-namespace Nethermind.Specs.ChainSpecStyle
+namespace Nethermind.Specs.ChainSpecStyle;
+
+public static class ChainSpecLoaderExtensions
 {
-    public static class ChainSpecLoaderExtensions
+
+    public static ChainSpec LoadEmbeddedOrFromFile(this IChainSpecLoader chainSpecLoader, string fileName, ILogger logger)
     {
-        public static ChainSpec LoadFromFile(this IChainSpecLoader chainSpecLoader, string filePath)
+        try
         {
-            filePath = filePath.GetApplicationResourcePath();
-            if (!File.Exists(filePath))
+            string resourceName = fileName;
+            if (!resourceName.Contains('/'))
             {
-                StringBuilder missingChainspecFileMessage = new($"Chainspec file does not exist {filePath}");
-                try
-                {
-                    missingChainspecFileMessage.AppendLine().AppendLine("Did you mean any of these:");
-                    string[] configFiles = Directory.GetFiles(Path.GetDirectoryName(filePath), "*.json");
-                    for (int i = 0; i < configFiles.Length; i++)
-                    {
-                        missingChainspecFileMessage.AppendLine($"  * {configFiles[i]}");
-                    }
-                }
-                catch (Exception)
-                {
-                    // do nothing - the lines above just give extra info and config is loaded at the beginning so unlikely we have any catastrophic errors here
-                }
-                finally
+                resourceName = $"chainspec/{resourceName}";
+            }
+            if (!resourceName.EndsWith(".json", StringComparison.OrdinalIgnoreCase))
+            {
+                resourceName = $"{resourceName}.json";
+            }
+            resourceName = resourceName.Replace('/', '.');
+            resourceName = $"Nethermind.Config.{resourceName}";
+            Assembly assembly = typeof(IConfig).Assembly;
+            string[] embeddedChainSpecFiles = assembly.GetManifestResourceNames();
+            if (!embeddedChainSpecFiles.Any(s => s.Equals(resourceName, StringComparison.OrdinalIgnoreCase)))
```

Comment:
Why enumerating through all resources instead of just calling `GetManifestResourceStream` and comparing to null?